### PR TITLE
StringChecker: leading hashbangs still report correct error line numbers

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

File: lib/string-checker.js [hunk 1]
```diff
@@ -76,6 +76,41 @@ StringChecker.prototype = {
...
+     * @return {Object}     the output AST
+     * @throws Error on invalid source code.
+     */
+    _parse: function(str) {
+        var hashbang = (str[0] === '#' && str[1] === '!');
+        var tree;
+
+        // Convert bin annotation to a comment
+        if (hashbang) {
+            str = '//' + str.substr(2, str.length);
```
Comment (mrjoelkemp) [inline lib/string-checker.js]: Why not just strip out the line (from `#!` to `\n` from `str`?

File: lib/string-checker.js [hunk 2]
```diff
@@ -76,6 +76,41 @@ StringChecker.prototype = {
...
 
     /**
+     * Parses source code into an AST
+     *
+     * @param  {String} str the source code to parse
+     * @return {Object}     the output AST
+     * @throws Error on invalid source code.
+     */
+    _parse: function(str) {
+        var hashbang = (str[0] === '#' && str[1] === '!');
```
Comment (mrjoelkemp) [inline lib/string-checker.js]: Same thing? `str.indexOf('#!') === 0`

File: lib/string-checker.js (hunk 1 above)
Comment (mikesherov) [inline lib/string-checker.js]: because then you're creating multiple line breaks which throws off `disallowMultipleLineBreaks`

**Comment by mikesherov** [comment]
@mrjoelkemp can I get a :+1: here.

File: lib/string-checker.js [hunk 3]
```diff
@@ -76,6 +76,41 @@ StringChecker.prototype = {
...
+     * @return {Object}     the output AST
+     * @throws Error on invalid source code.
+     */
+    _parse: function(str) {
+        var hashbang = str.indexOf('#!') === 0;
+        var tree;
+
+        // Convert bin annotation to a comment
+        if (hashbang) {
+            str = '//' + str.substr(2);
```
Comment (markelog) [inline lib/string-checker.js]: Could that conflict with `requireSpaceAfterLineComment ` rule?

**Comment by mrjoelkemp** [comment]
:+1: assuming Oleg's comment is not applicable.

File: lib/string-checker.js (hunk 3 above)
Comment (markelog) [inline lib/string-checker.js]: https://github.com/jscs-dev/node-jscs/pull/941/files#diff-c4ec64ef009f51578bddc596b5b6ddc2R107

I see, okay :-)

File: test/rules/disallow-multiple-line-breaks.js [hunk 4]
```diff
@@ -3,29 +3,37 @@ var assert = require('assert');
...
         assert(checker.checkString('x = 1;\n\ny = 2').isEmpty());
     });
+
     it('should not report single line break when first line starts with #!', function() {
-        checker.configure({ disallowMultipleLineBreaks: true });
         assert(checker.checkString('#!/usr/bin/env node\nx = 1;').isEmpty());
         assert(checker.checkString('#!/usr/bin/env node\n\nx = 1;').isEmpty());
     });
+
+    it('should report error in the correct location when first line starts with #!', function() {
```
Comment (markelog) [inline test/rules/disallow-multiple-line-breaks.js]: This pull for the `string-checker` module, but test for the rule, could we add one to for the relevant module too?

Like here - https://github.com/jscs-dev/node-jscs/blob/30049c1da94b590b783c1ba948da1d639b8c06bc/test/string-checker.js#L38

**Comment by mikesherov** [comment]
Closed by 0253717989b475999e2deea19f124ff1cbb502fd

**Pull request closed without merging**